### PR TITLE
Replace `MatrixClient.isRoomEncrypted` by `MatrixClient.CryptoApi.isEncryptionEnabledInRoom` in RoomView

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] I have licensed the changes to Element by completing the [Contributor License Agreement (CLA)](https://cla-assistant.io/element-hq/element-web)

### CODE REVIEW

File: src/components/structures/RoomView.tsx [hunk 1]
```diff
@@ -2367,34 +2403,37 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
...
-                membersLoaded={this.state.membersLoaded}
-                permalinkCreator={this.permalinkCreator}
-                resizeNotifier={this.props.resizeNotifier}
-                showReactions={true}
-                layout={this.state.layout}
-                editState={this.state.editState}
-            />
-        );
+        let messagePanel: JSX.Element | undefined;
+        if (!isRoomEncryptionLoading) {
```
Comment (florianduros) [inline src/components/structures/RoomView.tsx:2407]: I'm not sure about this one. The value of `this.state.showUrlPreview` needs to know if the room is encrypted.
One solution is to check if both settings of the url preview are the same but to do that we need the roomId which is also an async process... So not very useful...

File: src/components/structures/RoomView.tsx [hunk 2]
```diff
@@ -914,6 +929,7 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
         const callState = call?.state;
         this.setState({
             callState,
+            isRoomEncrypted: await this.getIsRoomEncrypted(),
```
Comment (t3chguy) [inline src/components/structures/RoomView.tsx:932]: This seems superfluous, what could have set `state.roomId` at this point in the flow?

File: src/components/structures/RoomView.tsx (hunk 1 above)
Comment (t3chguy) [reply src/components/structures/RoomView.tsx:2407]: Yeah this isn't right, the callback from setting `isRoomEncrypted` fires the method which sets the state necessary for URL previews, which means this has the chance to flicker if moving encrypted⊕!encrypted - they should be set in the same `setState` batch so they don't have this issue

File: playwright/e2e/crypto/decryption-failure-messages.spec.ts [hunk 3]
```diff
@@ -67,6 +67,9 @@ test.describe("Cryptography", function () {
             await page.locator(".mx_AuthPage").getByRole("button", { name: "I'll verify later" }).click();
             await app.viewRoomByName("Test room");
 
+            // In this case, the call to cryptoApi.isEncryptionEnabledInRoom is taking a long time to resolve
+            await page.waitForTimeout(4000);
```
[the remark this answers is not in the document]
Comment (florianduros) [reply playwright/e2e/crypto/decryption-failure-messages.spec.ts:71]: Top open a room in a regular case, it's not taking 4 seconds.  I tested manually and the other tests are running fine.

This test is going through already multiple issues (waiting 10sec for keys to be backed up)

**Review by t3chguy — APPROVED** [review]

**Pull request merged**